src/vm/excep.cpp

```diff
@@ -5683,18 +5684,28 @@ LONG __stdcall COMUnhandledExceptionFilter(     // EXCEPTION_CONTINUE_SEARCH or
     //
     // 4) A corrupting exception may become unhandled.
     //
-    // This is applicable to CoreCLR as well. We wont enter the UEF (and thus come here) on Silverlight,
-    // but we could when being used by WLC as we register the UEF for them.
+    // This is not applicable to CoreCLR, as this unhandled exception filter is always set up, and all hardware exceptions in
+    // managed code, including those that are not process-corrupting, such as integer division by zero, will end up here.
 
     // Assert these conditions here - we shouldnt be here for any other unhandled exception processing.
     Thread *pThread = GetThread();
-    _ASSERTE((pThread == NULL) || // condition 3
-             (pThread->GetExceptionState()->IsExceptionInProgress() &&
-              pThread->GetExceptionState()->GetFlags()->ReversePInvokeEscapingException()) || // condition 2
-             ((!ExecutionManager::IsManagedCode((PCODE)pExceptionInfo->ExceptionRecord->ExceptionAddress))) // condition 1
-             CORRUPTING_EXCEPTIONS_ONLY(||(CEHelper::IsProcessCorruptedStateException(pExceptionInfo->ExceptionRecord->ExceptionCode)) ||) // condition 4
-             CORRUPTING_EXCEPTIONS_ONLY((CEHelper::IsProcessCorruptedStateException(pThread->GetThrowable())))); // condition 4
+    if ((pThread != NULL) && // condition 3
+        !(pThread->GetExceptionState()->IsExceptionInProgress() &&
+            pThread->GetExceptionState()->GetFlags()->ReversePInvokeEscapingException()) && // condition 2
+        (ExecutionManager::IsManagedCode((PCODE)pExceptionInfo->ExceptionRecord->ExceptionAddress))) // condition 1
+    {
+#ifdef FEATURE_CORRUPTING_EXCEPTIONS
+        if (!CEHelper::IsProcessCorruptedStateException(pExceptionInfo->ExceptionRecord->ExceptionCode)) // condition 4
+        {
+            GCX_COOP();
+            _ASSERTE(CEHelper::IsProcessCorruptedStateException(pThread->GetThrowable())); // condition 4
+        }
+#else // !FEATURE_CORRUPTING_EXCEPTIONS
+        _ASSERTE(false);
```

Comment:
What is CSE?

Suppose FEATURE_CORRUPTING_EXCEPTIONS is off. On the left (before my changes), the code looks like:
```c#
_ASSERTE(A || B || C);
```

With my changes on the right, the code looks like:
```c#
if(!A && !B && !C)
{
    _ASSERTE(false);
}
```

I guess I'm missing something, as those should be equivalent.